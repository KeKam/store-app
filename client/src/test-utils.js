import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';

import rootReducer from './redux/root-reducer';

// const render = (
//   component,
//   {
//     initialState,
//     store = createStore(rootReducer, initialState),
//     ...renderOptions
//   } = {}
// ) => {
//   return {
//     ...rtlRender(<Provider store={store}>{component}</Provider>),
//     store,
//   };
//   // const Wrapper = ({ children }) => {
//   //   return <Provider store={store}>{children}</Provider>;
//   // };
//   // return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
// };

const render = (component, { initialState } = {}) => {
  const actions = [];
  const observerMiddleware = () => (next) => (action) => {
    actions.push(action);
    return next(action);
  };
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(observerMiddleware)
  );
  const utils = {
    dispatch(action) {
      return store.dispatch(action);
    },
    getDispatchedActions() {
      return actions;
    },
    getState() {
      return store.getState();
    },
  };
  return {
    ...rtlRender(<Provider store={store}>{component}</Provider>),
    ...utils,
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { render };
