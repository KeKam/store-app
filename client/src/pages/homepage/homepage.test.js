import React from 'react';
import '@testing-library/jest-dom';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { render, screen } from '../../test-utils';
import HomePage from './homepage';

const history = createMemoryHistory();

describe('HomePage component', () => {
  it('should render Homepage component', () => {
    const { asFragment } = render(
      <Router history={history}>
        <HomePage />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('SPRING')).toBeInTheDocument();
    expect(screen.getByText('SUMMER')).toBeInTheDocument();
    expect(screen.getByText('AUTUMN')).toBeInTheDocument();
    expect(screen.getAllByTestId('image')[0]).toBeInTheDocument();
    expect(screen.getAllByText('SEE COLLECTION')[0]).toBeInTheDocument();
  });
});
