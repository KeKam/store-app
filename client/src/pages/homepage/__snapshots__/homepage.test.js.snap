// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should render Homepage component 1`] = `
<DocumentFragment>
  <div
    class="sc-bZQynM bsmBBS"
  >
    <div
      class="sc-EHOje fpJsVf"
    >
      <div
        class="sc-bdVaJa hJLFdt"
        data-testid="item"
      >
        <div
          class="sc-bwzfXH cWENaq"
          data-testid="image"
        />
        <div
          class="sc-htpNat fMwJqu"
        >
          <div
            class="sc-bxivhb hQYITJ"
            data-testid="title"
          >
            SPRING
          </div>
          <div
            class="sc-ifAKCX jbyhsX"
          >
            SEE COLLECTION
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa hJLFdt"
        data-testid="item"
      >
        <div
          class="sc-bwzfXH dluYIx"
          data-testid="image"
        />
        <div
          class="sc-htpNat fMwJqu"
        >
          <div
            class="sc-bxivhb hQYITJ"
            data-testid="title"
          >
            SUMMER
          </div>
          <div
            class="sc-ifAKCX jbyhsX"
          >
            SEE COLLECTION
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa hJLFdt"
        data-testid="item"
      >
        <div
          class="sc-bwzfXH bkerVt"
          data-testid="image"
        />
        <div
          class="sc-htpNat fMwJqu"
        >
          <div
            class="sc-bxivhb hQYITJ"
            data-testid="title"
          >
            AUTUMN
          </div>
          <div
            class="sc-ifAKCX jbyhsX"
          >
            SEE COLLECTION
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa hJLFdt"
        data-testid="item"
      >
        <div
          class="sc-bwzfXH plgPp"
          data-testid="image"
        />
        <div
          class="sc-htpNat fMwJqu"
        >
          <div
            class="sc-bxivhb hQYITJ"
            data-testid="title"
          >
            WINTER
          </div>
          <div
            class="sc-ifAKCX jbyhsX"
          >
            SEE COLLECTION
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
